wandb/run_manager.py

```diff
@@ -140,32 +140,59 @@ def __init__(self, file_path, save_name, api, file_pusher, *args, **kwargs):
     def on_created(self):
         self.on_modified()
 
-    def on_modified(self):
-        current_time = time.time()
-        current_size = os.path.getsize(self.file_path)
+    @property
+    def current_size(self):
+        return os.path.getsize(self.file_path)
 
+    def on_modified(self):
         # Don't upload anything if it's zero size.
-        if current_size == 0:
-            return
+        if self.current_size == 0:
+            return 0
 
         if self._last_uploaded_time:
             # Check rate limit by time elapsed
-            time_elapsed = current_time - self._last_uploaded_time
+            time_elapsed = time.time() - self._last_uploaded_time
             if time_elapsed < self.RATE_LIMIT_SECONDS:
-                return
+                return time_elapsed
             # Check rate limit by size increase
-            size_increase = current_size / float(self._last_uploaded_size)
+            size_increase = self.current_size / float(self._last_uploaded_size)
             if size_increase < self.RATE_LIMIT_SIZE_INCREASE:
-                return
+                return time_elapsed
 
-        self._last_uploaded_time = current_time
-        self._last_uploaded_size = current_size
-        self._file_pusher.file_changed(
-            self.save_name, self.file_path, copy=True)
+        self.save_file()
+        return 0
 
     def finish(self):
         self._file_pusher.file_changed(self.save_name, self.file_path)
 
+    def save_file(self):
+        self._last_uploaded_time = time.time()
+        self._last_uploaded_size = self.current_size
+        self._file_pusher.file_changed(
+            self.save_name, self.file_path, copy=True)
+
+
+class FileEventHandlerThrottledOverwriteMinWait(FileEventHandlerThrottledOverwrite):
+    TEN_MB =     10000000
+    HUNDRED_MB = 100000000
+    ONE_GB =     1000000000
+
+    def min_wait_for_size(self, size):
+        if self.current_size < self.TEN_MB:
+            return 60
+        elif self.current_size < self.HUNDRED_MB:
+            return 5 * 60
+        elif self.current_size < self.ONE_GB:
+            return 10 * 60
+        else:
+            return 20 * 60
+
+    def on_modified(self):
```

Comment:
Cant this potentially miss updates? Like we drop one because not enough time has elapsed, then the run ends. You need to implement on_finished() or whatever probably.

Also, I'm not sure how all this stuff behaves when the file is being written to while we're reading it.

Reply:
We use copy=True when we push the file.  Because this inherits from Gabes file class it calls push on finish.

Reply:
ok